src/field_set.ts

```diff
@@ -10,5 +10,5 @@ export interface FieldSet {
         | Collaborator
         | ReadonlyArray<Collaborator>
         | ReadonlyArray<string>
-        | ReadonlyArray<Attachment>;
+        | ReadonlyArray<Attachment | CreateAttachment | AttachmentReference>;
```

Comment:
This is the only change I'm not sure about. 

Without it, we get TS errors when passing fields with type either`AttachmentReference` or `CreateAttachment` to the `create()` or `update()` methods because the result of `PartialFields`  has both fields' types and the indexed type from `FieldSet`:

```typescript
interface MyFields extends FieldSet {
  name: string;
  files: Attachment[]
}

type PartialMyFields = PartialFields<MyFields>;

/*
The result type of PartialMyFields is
{
  [x: string]:
      | undefined
      | string
      | number
      | boolean
      | Collaborator
      | ReadonlyArray<Collaborator>
      | ReadonlyArray<string>
      | ReadonlyArray<Attachment | CreateAttachment | AttachmentReference>;
  name: string;
  files: (AttachmentReference | CreateAttachment)[]
}
*/
```

Feel free to suggest a better solution.